src/Limiters/quasimonotone.jl

```diff
@@ -0,0 +1,348 @@
+    (_, _, _, Nv, Nh) = size(q_bounds_nbr)
+    for h in 1:Nh
+        for v in 1:Nv
+            slab_q_bounds = slab(q_bounds, v, h)
+            q_min = slab_q_bounds[1]
+            q_max = slab_q_bounds[2]
+            for h_nbr in Topologies.local_neighboring_elements(topology, h)
+                slab_q_bounds = slab(q_bounds, v, h_nbr)
+                q_min = rmin(q_min, slab_q_bounds[1])
+                q_max = rmax(q_max, slab_q_bounds[2])
+            end
+            slab_q_bounds_nbr = slab(q_bounds_nbr, v, h)
+            slab_q_bounds_nbr[1] = q_min
+            slab_q_bounds_nbr[2] = q_max
+        end
+    end
+    return nothing
+end
+
+function compute_neighbor_bounds_ghost!(
+    limiter::QuasiMonotoneLimiter,
+    topology::Topologies.AbstractTopology,
+)
+    q_bounds_nbr = limiter.q_bounds_nbr
+    (_, _, _, Nv, Nh) = size(q_bounds_nbr)
+    if limiter.ghost_buffer isa Spaces.GhostBuffer
+        q_bounds_ghost = limiter.ghost_buffer.recv_data
+
+        for h in 1:Nh
+            for v in 1:Nv
+                slab_q_bounds = slab(q_bounds_nbr, v, h)
+                q_min = slab_q_bounds[1]
+                q_max = slab_q_bounds[2]
+                for gidx in Topologies.ghost_neighboring_elements(topology, h)
+                    ghost_slab_q_bounds = slab(q_bounds_ghost, v, gidx)
+                    q_min = rmin(q_min, ghost_slab_q_bounds[1])
+                    q_max = rmax(q_max, ghost_slab_q_bounds[2])
+                end
+                slab_q_bounds_nbr = slab(q_bounds_nbr, v, h)
+                slab_q_bounds_nbr[1] = q_min
+                slab_q_bounds_nbr[2] = q_max
+            end
+        end
+    end
+    return nothing
+end
+
+"""
+    compute_bounds!(limiter::QuasiMonotoneLimiter, ρq::Field, ρ::Field)
+
+Compute the tracer bounds for the tracer concentration `q` based on the fields `ρq` and `ρ`.
+"""
+function compute_bounds!(
+    limiter::QuasiMonotoneLimiter,
+    ρq::Fields.Field,
+    ρ::Fields.Field,
+)
+    topology = Spaces.topology(axes(ρq))
+    compute_bounds!(
+        limiter,
+        Fields.field_values(ρq),
+        Fields.field_values(ρ),
+        topology,
+    )
+end
+function compute_bounds!(
+    limiter::QuasiMonotoneLimiter,
+    ρq,
+    ρ,
+    topology::Topologies.AbstractTopology,
+)
+    compute_element_bounds(limiter, ρq, ρ)
+    if limiter.ghost_buffer isa Spaces.GhostBuffer
+        Spaces.fill_send_buffer!(
+            topology,
+            limiter.q_bounds,
+            limiter.ghost_buffer,
+        )
+        ClimaComms.start(limiter.ghost_buffer.graph_context)
+    end
+    compute_neighbor_bounds_local!(limiter, topology)
+    if limiter.ghost_buffer isa Spaces.GhostBuffer
+        ClimaComms.finish(ghost_buffer.graph_context)
+        compute_neighbor_bounds_ghost!(limiter, topology)
+    end
+end
+
+
+
+
+"""
+    apply_limiter!(ρq, ρ, limiter::QuasiMonotoneLimiter)
+
+Apply the computed bounds of the tracer concentration (`q`) in the limiter to `ρq`.
```

Comment:
Does the following sound more what we are doing?
```suggestion
Apply the limiter on the tracer density  `ρq`, using the computed desired bounds and density, `ρ`, as an optimal weight.
```